src/v/features/feature_table.h

```diff
@@ -57,6 +57,7 @@ enum class feature : std::uint64_t {
     membership_change_controller_cmds = 1ULL << 22U,
     controller_snapshots = 1ULL << 23U,
     cloud_storage_manifest_format_v2 = 1ULL << 24U,
+    log_prefix_truncation = 1ULL << 25U,
```

Comment:
This naming is confusing: prefix truncation is not new, this particular mechanism is.  If you want to avoid calling it delete_records, then something like "explicit_prefix_truncation" might make more sense.  But it would also be fine just to call this "delete_records".